core/chain_makers.go

```diff
@@ -61,15 +61,14 @@ type BlockGen struct {
 // SetCoinbase sets the coinbase of the generated block.
 // It can be called at most once.
 func (b *BlockGen) SetCoinbase(addr common.Address) {
-	if b.coinbase != nil {
+	if b.gasPool != nil {
 		if len(b.txs) > 0 {
 			panic("coinbase must be set before adding transactions")
 		}
 		panic("coinbase can only be set once")
 	}
 	b.header.Coinbase = addr
-	b.coinbase = b.statedb.GetOrNewStateObject(addr)
-	b.coinbase.SetGasLimit(b.header.GasLimit)
+	b.gasPool = new(GasPool).AddGas(b.header.GasLimit)
```

Comment:
The relation between these two is that they both need to be set before adding/running any transactions. I'll try to make that clearer.